frontend/src/metabase/lib/utils.js

```diff
@@ -135,36 +135,58 @@ const MetabaseUtils = {
     return JSON.parse(JSON.stringify(a));
   },
 
-  // this should correctly compare all version formats Metabase uses, e.x.
-  // 0.0.9, 0.0.10-snapshot, 0.0.10-alpha1, 0.0.10-rc1, 0.0.10-rc2, 0.0.10-rc10
-  // 0.0.10, 0.1.0, 0.2.0, 0.10.0, 1.1.0
-  compareVersions(aVersion, bVersion) {
-    if (!aVersion || !bVersion) {
-      return null;
-    }
-
+  /**
+   * Converts a metabase version to a list of numeric components, it converts pre-release
+   * components to numbers and pads the numeric part to 4 numbers to make comparison easier
+   * @param {string} version
+   * @returns {number[]}
+   */
+  versionToNumericComponents(version) {
     const SPECIAL_COMPONENTS = {
       snapshot: -4,
       alpha: -3,
       beta: -2,
       rc: -1,
     };
 
-    const getComponents = x =>
-      // v1.2.3-BETA1
-      x
-        .toLowerCase()
-        // v1.2.3-beta1
-        .replace(/^v/, "")
-        // 1.2.3-beta1
-        .split(/[.-]*([0-9]+)[.-]*/)
-        .filter(c => c)
-        // ["1", "2", "3", "beta", "1"]
-        .map(c => SPECIAL_COMPONENTS[c] || parseInt(c, 10));
-    // [1, 2, 3, -2, 1]
-
-    const aComponents = getComponents(aVersion);
-    const bComponents = getComponents(bVersion);
+    const regex =
+      /v?(?<ossOrEE>\d+)\.?(?<major>\d+)?\.?(?<minor>\d+)?\.?(?<patch>\d+)?-?(?<label>\D+)?(?<build>\d+)?/;
```

Comment:
this is clever, but we should consider just using the battle-tested

https://www.npmjs.com/package/semver

Reply:
@iethree We tried to use that but we're not really 100% semver standard compatible :( 
See https://github.com/metabase/metabase/pull/33671#issuecomment-1705431936